components/in_memory_engine/src/background.rs

```diff
@@ -1330,10 +1332,37 @@ impl RunnableWithTimer for BackgroundRunner {
         IN_MEMORY_ENGINE_MEMORY_USAGE.set(mem_usage as i64);
 
         let mut count_by_state = [0; RegionState::COUNT];
+        let mut oldest_safe_point = u64::MAX;
+        let mut newest_safe_point = u64::MIN;
         {
             let regions_map = self.core.engine.region_manager().regions_map.read();
             for m in regions_map.regions().values() {
                 count_by_state[m.get_state() as usize] += 1;
+                if m.get_state() == RegionState::Active && m.safe_point() != 0 {
+                    oldest_safe_point = u64::min(oldest_safe_point, m.safe_point());
+                    newest_safe_point = u64::max(newest_safe_point, m.safe_point());
+                }
+            }
+        }
+
+        if oldest_safe_point != u64::MAX {
+            IN_MEMORY_ENGINE_OLDEST_SAFE_POINT.set(oldest_safe_point as i64);
+            IN_MEMORY_ENGINE_NEWEST_SAFE_POINT.set(newest_safe_point as i64);
+            if let Ok(tikv_safe_point) = block_on(self.pd_client.get_gc_safe_point()) {
```

Comment:
Prefer `block_on_timeout` in case `get_gc_safe_point` blocks forever.

Reply:
Ok